praw/objector.py

```diff
@@ -36,13 +44,17 @@ def parse_error(cls, data):
         return APIException(*errors[0])
 
     @classmethod
-    def check_error(cls, data):
+    def check_error(
+        cls, data: Union[List[Any], Dict[str, Dict[str, str]]]
+    ) -> NoReturn:
```

Comment:
I believe usage of `NoReturn` here is incorrect. Per [the documentation](https://docs.python.org/3/library/typing.html#typing.NoReturn):

> Special type indicating that a function never returns.

The `check_error` function *sometimes* never returns (when an exception is raised) but usually returns `None` (when no exception is raised). Type annotations should be updated to reflect this. I'm not familiar with type annotations, but perhaps a `Union[NoneType, NoReturn]` would be appropriate.

Reply:
I decided to just not use NoReturn so this should be phased out as well. I’m on a phone now so I’ll make a PR tomorrow.

Reply:
Good catch @jarhill0. To be honest, I haven't really scrutinized these PRs at all. I figure it's all an improvement, even if it's not 100% perfect.